libs/sdk-core/src/models.rs

```diff
@@ -641,6 +645,180 @@ pub struct ClosedChannelPaymentDetails {
+    /// could be invalid. Therefore, the [OpeningFeeParamsMenu] should always be initialized on-the-fly.
+    pub fn try_from(vals: Vec<grpc::OpeningFeeParams>) -> Result<Self> {
+        let temp = Self {
+            vals: vals
+                .into_iter()
+                .map(|ofp| ofp.into())
+                .collect::<Vec<OpeningFeeParams>>(),
+        };
+        temp.validate().map(|_| temp)
+    }
+
+    fn validate(&self) -> Result<()> {
+        // opening_fee_params_menu fees must be in ascending order
+        let is_ordered = self.vals.windows(2).all(|ofp| {
+            let larger_min_msat_fee = ofp[0].min_msat < ofp[1].min_msat;
+            let equal_min_msat_fee = ofp[0].min_msat == ofp[1].min_msat;
+
+            let larger_proportional = ofp[0].proportional < ofp[1].proportional;
+            let equal_proportional = ofp[0].proportional == ofp[1].proportional;
+
+            (larger_min_msat_fee && equal_proportional)
+                || (equal_min_msat_fee && larger_proportional)
+                || (larger_min_msat_fee && larger_proportional)
+        });
+        ensure!(is_ordered, "Validation failed: fee params are not ordered");
+
+        // Menu must not contain any item with `valid_until` in the past
+        let is_expired = self.vals.iter().any(|ofp| match ofp.valid_until_date() {
+            Ok(valid_until) => Utc::now() > valid_until,
+            Err(_) => {
+                warn!("Failed to parse valid_until for OpeningFeeParams: {ofp:?}");
+                false
+            }
+        });
+        ensure!(!is_expired, "Validation failed: expired fee params found");
+
+        Ok(())
+    }
+
+    pub(crate) fn get_cheapest_opening_fee_params(&self) -> Result<OpeningFeeParams> {
+        self.vals.first().cloned().ok_or(anyhow!(
+            "The LSP doesn't support opening new channels: Dynamic fees menu contains no values"
+        ))
+    }
+
+    pub(crate) fn get_longest_valid_opening_fee_params(&self) -> Result<OpeningFeeParams> {
+        // Find the fee params that are valid for at least 48h
+        let now = Utc::now();
+        let duration_48h = chrono::Duration::hours(48);
+        let valid_min_48h: Vec<OpeningFeeParams> = self
+            .vals
+            .iter()
+            .filter(|ofp| match ofp.valid_until_date() {
+                Ok(valid_until) => valid_until - now > duration_48h,
+                Err(_) => {
+                    warn!("Failed to parse valid_until for OpeningFeeParams: {ofp:?}");
+                    false
+                }
+            })
+            .cloned()
+            .collect();
+
+        // Of those, return the first, which is the cheapest
+        // (sorting order of fee params list was checked when the menu was initialized)
+        valid_min_48h.first().cloned().ok_or(anyhow!(
+            "The LSP doesn't support opening new channels: Dynamic fees menu contains no values"
+        ))
```

Comment:
There's a slight difference between the menu being empty (LSP doesn't support opening channels), and the menu being empty after filtering for 48h duration (LSP doesn't support params with a duration > 48h). Maybe we can distinguish in error messages here.

Reply:
Done: 423ac35d161e2c5c85a8d61ec5ab28988f34af82